tests/PHPStan/Type/CallableTypeTest.php

```diff
@@ -0,0 +1,84 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type;
+
+use PHPStan\TrinaryLogic;
+
+class CallableTypeTest extends \PHPStan\TestCase
+{
+
+	public function dataIsSubsetOf(): array
+	{
+		return [
+			[
+				new CallableType(),
+				new CallableType(),
+				TrinaryLogic::createYes(),
+			],
+			[
+				new CallableType(),
+				new StringType(),
+				TrinaryLogic::createMaybe(),
+			],
+			[
+				new CallableType(),
+				new IntegerType(),
+				TrinaryLogic::createNo(),
+			],
+			[
+				new CallableType(),
+				new CommonUnionType([new CallableType(), new NullType()]),
+				TrinaryLogic::createYes(),
+			],
+			[
+				new CallableType(),
+				new CommonUnionType([new StringType(), new NullType()]),
+				TrinaryLogic::createMaybe(),
+			],
+			[
+				new CallableType(),
+				new CommonUnionType([new IntegerType(), new NullType()]),
+				TrinaryLogic::createNo(),
+			],
+			[
+				new CallableType(),
+				new IntersectionType([new CallableType()]),
+				TrinaryLogic::createYes(),
+			],
+			[
+				new CallableType(),
+				new IntersectionType([new StringType()]),
+				TrinaryLogic::createMaybe(),
+			],
+			[
+				new CallableType(),
+				new IntersectionType([new IntegerType()]),
+				TrinaryLogic::createNo(),
+			],
+			[
+				new CallableType(),
+				new IntersectionType([new CallableType(), new StringType()]),
+				TrinaryLogic::createMaybe(),
+			],
+			[
+				new CallableType(),
+				new IntersectionType([new CallableType(), new ObjectType('Unknown')]),
+				TrinaryLogic::createMaybe(),
+			],
+		];
+	}
+
+	/**
+	 * @dataProvider dataIsSubsetOf
+	 * @param CallableType $callableType
+	 * @param Type $otherType
+	 * @param int $expectedResult
+	 */
+	public function testIsSubsetOf(CallableType $callableType, Type $otherType, TrinaryLogic $expectedResult)
+	{
+		$this->createBroker();
+		$this->assertEquals($expectedResult, $otherType->isSupersetOf($callableType));
```

Comment:
Do not rely on magic comparison of objects, use assertTrue/assertFalse on TrinaryLogic::equals().

Reply:
I've reworked all tests to use data providers and compare trinary logic values in a way that produces very readable error messages, e.g.

![image](https://user-images.githubusercontent.com/175109/30188233-b6fbac0a-942d-11e7-8757-774e5990e7f1.png)